t/ust.t

```diff
@@ -0,0 +1,275 @@
+                'versionHex'        => '20000000',
+                'nTx'               => 38,
+                'previousblockhash' => '000000009ec12186d1406c6dbcdbb7714c9d8e7eb0c8f661c3df0966685ae226',
+                'confirmations'     => 1,
+                'strippedsize'      => 12219
+            };
+
+            return Future->done($getblock);
+        },
+        list_by_addresses => sub {
+            my ($self, $address) = @_;
+            my $result;
+            if ($address eq 'mgVxUb5mYJkfoo4w2JBVBcWaP5bqtLroTr') {
+                $result = [{
+                        txids   => ['19e5bda81d0a588cbf46072fb6da01685a51082d4b94428e09f491ad8a111bc7'],
+                        address => 'mgVxUb5mYJkfoo4w2JBVBcWaP5bqtLroTr'
+                    }];
+            }
+
+            else { $result = []; }
+
+            return Future->done($result);
+        },
+        get_transaction => sub {
+            my ($self, @params) = @_;
+            my $omni_gettransaction = {
+                'confirmations' => 1,
+                'version'       => 0,
+                'type_int'      => 4,
+                'subsends'      => [{
+                        'amount'     => '56.00000000',
+                        'propertyid' => 2147484941,
+                        'divisible'  => 1
+                    }
+                ],
+                'ecosystem'        => 'test',
+                'positioninblock'  => 36,
+                'type'             => 'Send All',
+                'referenceaddress' => 'mgVxUb5mYJkfoo4w2JBVBcWaP5bqtLroTr',
+                'ismine'           => 1,
+                'block'            => 1666293,
+                'fee'              => '0.00000245',
+                'valid'            => 1,
+                'blocktime'        => 1582186008,
+                'sendingaddress'   => 'mvaK72Z5EVYJtptWrD2nCSZd1nqEx5B469',
+                'blockhash'        => '0000000027072628ae58009d904e586cce12f2c68d3a501cbb0a781555bcb5d2',
+                'txid'             => '1b007afef8d271861dccd5b6bfe2f7e45a481c9a2f39b9f063e43e7d26375ec7'
+            };
+
+            return Future->done($omni_gettransaction);
+        });
+
+    my $subscription_source = $subscription_client->source;
+    $subscription_source->each(
+        sub {
+            my $transaction = shift;
+            is $transaction->{currency}, 'UST',     'Currency code is correct';
+            is $transaction->{amount},   "56",      'Amount is correct';
+            is $transaction->{type},     'receive', 'Transaction Type is correct';
+            is_deeply $transaction, $expected_transaction, "$currency_code Omni-Send All Transaction is emitted correctly.";
+            $subscription_source->finish();
+        });
+
+    my $emitted_transaction = $subscription_client->hashblock('0000000027072628ae58009d904e586cce12f2c68d3a501cbb0a781555bcb5d2')->get;
+
+    $mocked_omni->unmock_all();
+    $mocked_rpc_omni->unmock_all();
+};
+
+done_testing();
```

Comment:
You should add a test for the internal transactions as well

Reply:
Added. 👍